nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -465,7 +474,39 @@ in {
     assertions = [
       {
         assertion = databaseActuallyCreateLocally -> (cfg.user == cfg.database.user);
-        message = ''For local automatic database provisioning (services.mastodon.database.createLocally == true) with peer authentication (services.mastodon.database.host == "/run/postgresql") to work services.mastodon.user and services.mastodon.database.user must be identical.'';
+        message = ''
+          For local automatic database provisioning (services.mastodon.database.createLocally == true) with peer
+            authentication (services.mastodon.database.host == "/run/postgresql") to work services.mastodon.user
+            and services.mastodon.database.user must be identical.
+        '';
+      }
+      {
+        assertion = !databaseActuallyCreateLocally -> (cfg.database.host != "/run/postgresql");
+        message = ''
+          <option>services.mastodon.database.host</option> needs to be set if
+            <option>services.mastodon.database.createLocally</option> is not enabled.
+        '';
+      }
+      {
+        assertion = !databaseActuallyCreateLocally -> (cfg.database.passwordFile != null);
```

Comment:
There are other ways of authenticating against a remote PostgreSQL instance, e.g. via TLS client certificates. This assertion would break such a setup.

Reply:
I couldn't find certificate settings in `services.postgresql`.

Reply:
Yeah it's a bit more complicated to configure, On my services I do it via `services.postgresql.authentication`. Also the PostgreSQL instance doesn't necessarily run NixOS at all. I still think that this assertion assumes too much.

Reply:
Delete these lines?

Reply:
Removed.